rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/activity/CovidPatientRegisterActivityTest.java

```diff
@@ -72,14 +76,14 @@ public void testLanguageSwitcherDialogShouldSetCorrectLocale() throws Exception
 
     private void verifyLocaleIsSaved(ListView listView, int position, String locale) {
         listView.performItemClick(null, position, listView.getItemIdAtPosition(position));
-        Assert.assertEquals(locale, LangUtils.getLanguage(RDTApplication.getInstance()));
-        Assert.assertEquals(locale, RDTApplication.getInstance().getResources().getConfiguration().locale.getLanguage());
+        Assert.assertEquals(locale, new Locale(LangUtils.getLanguage(covidPatientRegisterActivity)).getLanguage());
     }
 
     @Test
     public void testRegisterLanguageSwitcherShouldVerifyLocaleIndex() throws Exception {
         verifySavedLanguageIndex(1);
-        LangUtils.saveLanguage(covidPatientRegisterActivity, locales[0]);
+        LangUtils.saveLanguage(covidPatientRegisterActivity, "en");
+        Utils.updateLocale(covidPatientRegisterActivity);
```

Comment:
If this test is supposed to verify the default language is set then I don't think we need this block. We should simply have one call to `verifySavedLanguageIndex(i);`

Reply:
if I remove the above code snippet so I can only verify the default locale index but if I need to verify another locale index then I need to update the locale first.

Reply:
Ok, we can keep it. Is there a reason "en" is hard-coded now instead of using `locales` array like before?

Reply:
removed hardcoded locale and get from the resource file